datafusion/core/src/physical_optimizer/pruning.rs

```diff
@@ -276,21 +383,21 @@ fn is_always_true(expr: &Arc<dyn PhysicalExpr>) -> bool {
 /// Handles creating references to the min/max statistics
 /// for columns as well as recording which statistics are needed
 #[derive(Debug, Default, Clone)]
-pub(crate) struct RequiredStatColumns {
+pub(crate) struct RequiredColumns {
```

Comment:
I renamed this to be more specific and since it is `crate` private it is not a breaking API change

One thing I did try was encoding the columns needed for literal guarantees in this structure, but I found the code was very specific to `min/max/count` statistics

Reply:
Yeah, when I worked on statistics where I only needed min and max, I did not see the need to to use the available struct that include a lot more info